src/wallet/wallet.cpp

```diff
@@ -6750,6 +6757,117 @@ bool CWallet::InitLoadWallet(const CChainParams& params, bool clearWitnessCaches
+                        lastUsedLine++;
+                        auto seedFpStr = line.substr(fingerprintPos + 12, std::string::npos);
+                        boost::trim(seedFpStr);
+                        uint256 seedFp;
+                        seedFp.SetHex(seedFpStr.c_str());
+                        fingerprint = seedFp;
+                    }
+
+                    // Syntax:
+                    //     num_accounts=INT
+                    //
+                    // Constrained to immediately follow fingerprint= (if present).
+                    //
+                    // If this line is not present, treat it as num_accounts=1.
+                    // If num_accounts=0, treat it as such (no automatic account recovery).
+                    auto numAccountsPos = line.find("num_accounts=");
+                    if (numAccountsPos != std::string::npos) {
+                        assert(recoveryPhrase.has_value());
+                        assert(lineNumber == (lastUsedLine + 1));
+                        lastUsedLine++;
+                        auto numAccountsStr = line.substr(numAccountsPos + 13, std::string::npos);
+                        boost::trim(numAccountsStr);
+                        numAccounts = static_cast<uint32_t>(std::stoul(numAccountsStr.c_str()));
+                    }
+                }
+            }
+
+            if (recoveryPhrase.has_value()) {
+                auto seed = MnemonicSeed::ForPhrase(
+                    language.value_or(MnemonicSeed::LanguageFromName("English").value()),
+                    recoveryPhrase.value());
+                if (seed.has_value()) {
+                    // Check against fingerprint, if available
+                    if (fingerprint.has_value()) {
+                        assert(seed.value().Fingerprint() == fingerprint.value());
+                    }
+                    walletInstance->InitWithSeed(seed.value());
+                    for (uint32_t acctId = 0; acctId < numAccounts; acctId++) {
+                        walletInstance->GenerateNewUnifiedSpendingKey();
+                        walletInstance->GenerateUnifiedAddress(acctId, CWallet::DefaultReceiverTypes(chainActive.Height()), std::nullopt);
+                    }
+                } else if (!language.has_value()) {
+                    return UIError(_("Incorrect language provided for the given recovery phrase. Check that the phrase is correct, and add a language=LANGUAGE_NAME line if the recovery phrase language is not English."));
+                } else {
+                    return UIError(_("Unable to generate a valid seed from the given recovery phrase and language."));
+                }
+                assert(walletInstance->HaveMnemonicSeed());
+            } else {
+                return UIError(_("Unable to parse mnemonic phrase from recovery file."));
+            }
+        } else {
+            return UIError(_("Wallet recover file not found."));
+        }
```

Comment:
In general I prefer to check for an error and return early, when the error-handling code is short, so that you don't have to look forward in the file for the matching error handling code (especially when there is multiple nesting like this).